db/substate_db.go

```diff
@@ -88,6 +91,48 @@ func (db *substateDB) GetSubstate(block uint64, tx int) (*substate.Substate, err
 	return rlpSubstate.ToSubstate(db.GetCode, block, tx)
 }
 
+func (db *substateDB) GetBlockSubstates(block uint64) map[int]*substate.Substate {
+	var err error
+
+	txSubstate := make(map[int]*substate.Substate)
+
+	prefix := SubstateDBBlockPrefix(block)
+
+	iter := db.backend.NewIterator(util.BytesPrefix(prefix), nil)
+	for iter.Next() {
+		key := iter.Key()
+		value := iter.Value()
+
+		b, tx, err := DecodeSubstateDBKey(key)
+		if err != nil {
+			panic(fmt.Errorf("record-replay: invalid substate key found for block %v: %v", block, err))
+		}
+
+		if block != b {
+			panic(fmt.Errorf("record-replay: GetBlockSubstates(%v) iterated substates from block %v", block, b))
+		}
+
+		rlpSubstate, err := rlp.Decode(value)
+		if err != nil {
+			panic(fmt.Errorf("cannot decode data into rlp block: %v, tx %v; %v", block, tx, err))
+		}
+
+		sbstt, err := rlpSubstate.ToSubstate(db.GetCode, block, tx)
+		if err != nil {
+			panic(fmt.Errorf("cannot decode data into substate: %v", err))
+		}
+
+		txSubstate[tx] = sbstt
+	}
+	iter.Release()
+	err = iter.Error()
+	if err != nil {
+		panic(err)
+	}
```

Comment:
Please return the errors instead of `panicking`

Reply:
Make sense, thanks